### ISSUE TITLE
[BUG] `ForecastingGridSearchCV` with `NaiveForecaster` produces inconsistent column names  :  userwarning

### ISSUE THREAD

**Issue by fkiraly** [sktime/sktime#3510]
### Discussed in https://github.com/sktime/sktime/discussions/3498

<div type='discussions-op-text'>

<sup>Originally posted by **Ajay-user** September 28, 2022</sup>
Hi , I'm new to sktime

I'm getting this user waring when using forecastingGridSearchCV 
**UserWarning**: y_pred and y_true do not have the same column index. This may indicate incorrect objects passed to the metric. Indices of y_true will be used for y_pred.

Is there any work-around ? or is it normal?

[sktime documentation link](https://www.sktime.org/en/stable/api_reference/auto_generated/sktime.forecasting.model_selection.ForecastingGridSearchCV.html
)

```python
from sktime.datasets import load_shampoo_sales
from sktime.forecasting.naive import NaiveForecaster
from sktime.forecasting.model_selection import ForecastingGridSearchCV, ExpandingWindowSplitter

y = load_shampoo_sales()

fh = [1,2,3]

cv = ExpandingWindowSplitter(
    start_with_window=True,
    fh=fh)

forecaster = NaiveForecaster()

param_grid = {"strategy" : ["last", "mean", "drift"]}

gscv = ForecastingGridSearchCV(
    forecaster=forecaster,
    param_grid=param_grid,
    cv=cv)

gscv.fit(y)
```

_UserWarning_: _y_pred and y_true do not have the same column index. This may indicate incorrect objects passed to the metric. Indices of y_true will be used for y_pred._</div>

**Comment by fkiraly** [comment]
I can confirm this on windows, python 3.9, current main

**Comment by GianlucaGuarroSonos** [comment]
I get the same warnings evaluating using the `ExpandingWindowSplitter`

```
cv = ExpandingWindowSplitter(
    step_length=12, fh=[1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12], initial_window=72
)

forecaster = Prophet(  
    seasonality_mode='multiplicative',
    n_changepoints=int(len(newusers_df) / 12),
    add_country_holidays={'country_name': 'UnitedStates'},
    yearly_seasonality=True
)

df = evaluate(forecaster=forecaster, y=y, X=X, cv=cv, strategy='refit', return_data=True)
```

<img width="1418" alt="image" src="https://user-images.githubusercontent.com/107063818/195135931-d4a67422-c4e2-4914-803c-e9857dcfe39e.png">

**Comment by fkiraly** [comment]
> It's worth noting that I do not get this warning when I use a metric like `mean_absolute_percentage_error`

Yes, that's explained because the input/output checks sit in the metrics class but not the metrics function.

**Comment by hazrulakmal** [comment]
I ran both scripts from @GianlucaGuarroSonos & Ajay-user but I didn't get to reproduce the errors on the latest version sktime `0.19.1`. Based on the given code examples, I suspect that this issue is related to #2846.

**Comment by fkiraly** [comment]
yes, the issue is resolved apparently, see #2846 for further explanation.

The code vignette no longer runs since `start_with_window` got deprecated , but removing it yields running code should trigger the issue if it still were there.
